MapboxDirections/MBRouteStep.swift

```diff
@@ -382,7 +436,11 @@ public class RouteStep: NSObject, NSSecureCoding {
     
     internal init(finalHeading: CLLocationDirection?, maneuverType: ManeuverType?, maneuverDirection: ManeuverDirection?, maneuverLocation: CLLocationCoordinate2D, name: String?, coordinates: [CLLocationCoordinate2D]?, json: JSONDictionary) {
         transportType = TransportType(description: json["mode"] as! String)
-        destinations = json["destinations"] as? String
+        
+        let road = Road(name: name, ref: json["ref"] as? String, destination: json["destinations"] as? String)
+        names = road.names
+        codes = road.codes ?? road.destinationCodes
```

Comment:
The idea here was to make it easier for applications to display a route shield for an upcoming ramp based on `codes` without having to consult a separate property. We’ve since moved away from the notion that MapboxDirections.swift should perform any sort of munging, so I’ll remove this fallback.

I’ve gone back and forth quite a bit about whether destination codes should be conflated with `destination` or live in a separate property. In OSM, `destination:ref` usually follows a different format and encodes more information than `ref`. For example, if I-80 travels over a ramp to go from one motorway to another, the `highway=motorway_link` way is tagged `ref=I 80` but may also be tagged `destination:ref=I-80 West`. The format is loose enough that I don’t feel confident making `destinationCodes` look like `codes`. But I guess it can’t hurt much.